### PR TITLE
TransferTask Proto Persistence

### PR DESCRIPTION
**Pull request by shawnhathaway**
- Converting TransferTask pogo usage into proto
- SQL - Move TransferTask thrift serialization to Proto
- Cassandra - Move from Frozen<transfer_task> to proto
- Serialization occurs at store level until ExecutionStore interface is created with datablobs

### CODE REVIEW

File: common/persistence/cassandra/cassandraPersistenceUtil.go [hunk 1]
```diff
@@ -995,19 +995,15 @@ func createTransferTasks(
 			targetDomainID = task.(*p.CancelExecutionTask).TargetDomainID
 			targetWorkflowID = task.(*p.CancelExecutionTask).TargetWorkflowID
 			targetRunID = task.(*p.CancelExecutionTask).TargetRunID
-			if targetRunID == "" {
```
Comment (alexshtin) [inline common/persistence/cassandra/cassandraPersistenceUtil.go]: Why did you remove this?

File: common/persistence/sql/blob.go [hunk 2]
```diff
@@ -231,13 +231,13 @@ func taskListInfoFromBlob(b []byte, proto string) (*sqlblobs.TaskListInfo, error
 	return result, thriftRWDecode(b, proto, result)
 }
 
-func transferTaskInfoToBlob(info *sqlblobs.TransferTaskInfo) (p.DataBlob, error) {
-	return thriftRWEncode(info)
+func TransferTaskInfoToBlob(info *persistenceblobs.TransferTaskInfo) (p.DataBlob, error) {
+	return protoRWEncode(info)
```
Comment (alexshtin) [inline common/persistence/sql/blob.go]: `RW` means read/write. Why do we have it here, as part of a func name?

File: common/persistence/cassandra/cassandraPersistenceUtil.go (hunk 1 above)
Comment (shawnhathaway) [reply common/persistence/cassandra/cassandraPersistenceUtil.go]: This was just used internally as a default value marker for Cassandra's frozen<transfer_task>.RunID which was a UUID. It would write a well known UUID if the RunID was not specified and then it would swap this back out on deserialization with an empty string. Now that RunID is exposed as a byte array itself, this seems unneeded and we can propagate nil if thats what the customer provides.

File: common/persistence/sql/blob.go (hunk 2 above)
Comment (shawnhathaway) [reply common/persistence/sql/blob.go]: 🤷‍♂ Just a quick rename, will change this when I move/cleanup blob.go

File: common/persistence/cassandra/cassandraPersistenceUtil.go (hunk 1 above)
Comment (shawnhathaway) [reply common/persistence/cassandra/cassandraPersistenceUtil.go]: See func createTransferTaskInfo() for the other side of this
